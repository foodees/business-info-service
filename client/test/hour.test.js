import React from 'react';
import Hours from '../src/hours.jsx';
import renderer from 'react-test-renderer';

const data ={
	"business_id": "07AZL5XenCQ_-op_onKLdw",
	"name": "BLT Steak",
	"neighborhood": "The Strip", 
	"address": "3645 Las Vegas Blvd S", 
	"city": "Las Vegas", 
	"state": "NV", 
	"postal_code": "89109", 
	"latitude": 36.113669, 
	"longitude": -115.168933, 
	"stars": 4.0, 
	"review_count": 482, 
	"is_open": 1, 
	"attributes": {
		"RestaurantsTableService": true, 
		"GoodForMeal": {
			"dessert": false, 
			"latenight": false, 
			"lunch": false, 
			"dinner": true, 
			"breakfast": false, 
			"brunch": false
		},
		"Alcohol": "full_bar", 
		"Caters": false, 
		"HasTV": true, 
		"RestaurantsGoodForGroups": true, 
		"NoiseLevel": "average", 
		"WiFi": "no", 
		"RestaurantsAttire": "casual", 
		"RestaurantsReservations": true, 
		"OutdoorSeating": false, 
		"BusinessAcceptsCreditCards": true, 
		"RestaurantsPriceRange2": 3, 
		"BikeParking": false, 
		"RestaurantsDelivery": false, 
		"Ambience": {
			"romantic": false, 
			"intimate": false, 
			"classy": false, 
			"hipster": false, 
			"divey": false, 
			"touristy": false, 
			"trendy": false, 
			"upscale": true, 
			"casual": false
		}, 
		"RestaurantsTakeOut": true, 
		"GoodForKids": false, 
		"WheelchairAccessible": true, 
		"BusinessParking": {
			"garage": true, 
			"street": false, 
			"validated": false, 
			"lot": false, 
			"valet": true
		}
	}, 
	"categories": ["Steakhouses", "American (Traditional)", "Restaurants"],
	"hours": {
		"Monday": "17:00-22:00",
		"Tuesday": "17:00-22:00", 
		"Friday": "17:00-22:30", 
		"Wednesday": "17:00-22:00", 
		"Thursday": "17:00-22:00", 
		"Sunday": "17:00-22:00", 
		"Saturday": "17:00-22:30"
	}
}

it('renders correctly', () => {
  const tree = renderer
    .create(<Hours infors={data}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('should have correct number of $', function () {
	const component = renderer.create(
		<Hours infors={data}/>
	)
	const tree = component.toTree();
	const instance = component.root
 	// expect(tree.rendered.rendered[0].rendered[2].rendered[0].rendered[0]).toBe("$$$");
 	expect(instance.findByProps({className: 'money'}).children[0]).toBe('$$$')
});




