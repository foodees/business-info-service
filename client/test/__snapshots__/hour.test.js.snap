// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="smalltable col-md-5"
  >
    <p>
      <i
        className="far fa-clock"
        style={
          Object {
            "color": "green",
          }
        }
      />
      <span
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        Today:
      </span>
       
      <span>
        <span>
          17:00-22:00
        </span>
        <span
          style={
            Object {
              "marginLeft": "10px",
            }
          }
        >
          <b
            style={
              Object {
                "color": "green",
              }
            }
          >
            Open Now
          </b>
        </span>
      </span>
    </p>
    <p
      style={
        Object {
          "borderTop": "1px solid #e6e6e6",
          "height": "1px",
        }
      }
    />
    <p
      style={
        Object {
          "marginLeft": "10px",
        }
      }
    >
      <span
        className="money"
      >
        $$$
      </span>
        Price range
      <b
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        <span>
          $31 - 60
        </span>
      </b>
    </p>
  </div>
  <h3
    className="hours"
  >
    Hours
  </h3>
  <table
    className="hoursTable"
  >
    <tbody>
      <tr>
        <th>
          <b>
            Mon
          </b>
        </th>
        <td>
          17:00-22:00
           
        </td>
      </tr>
      <tr>
        <th>
          <b>
            Tue
          </b>
        </th>
        <td>
          17:00-22:00
           
        </td>
      </tr>
      <tr>
        <th>
          <b>
            Wed
          </b>
        </th>
        <td>
          17:00-22:00
           
          <b
            style={
              Object {
                "color": "green",
              }
            }
          >
            Open Now
          </b>
        </td>
      </tr>
      <tr>
        <th>
          <b>
            Thu
          </b>
        </th>
        <td>
          17:00-22:00
           
        </td>
      </tr>
      <tr>
        <th>
          <b>
            Fri
          </b>
        </th>
        <td>
          17:00-22:30
           
        </td>
      </tr>
      <tr>
        <th>
          <b>
            Sat
          </b>
        </th>
        <td>
          17:00-22:30
           
        </td>
      </tr>
      <tr>
        <th>
          <b>
            Sun
          </b>
        </th>
        <td>
          17:00-22:00
           
        </td>
      </tr>
    </tbody>
  </table>
  <p>
    <i
      className="fas fa-pencil-alt"
    />
        
    <a
      className="typeFood"
      href="https://www.yelp.com/biz_attribute?biz_id=07AZL5XenCQ_-op_onKLdw"
    >
      Edit business info
    </a>
  </p>
</div>
`;
